src/index-node.ts

```diff
@@ -28,9 +28,21 @@ export class SiweClient extends SiweImpl {
 
   async _extractCookies(headers?: Headers): Promise<void> {
     const cookieRecord: Record<string, string> = {}
-    const headerSetCookies = (headers as any).raw()[
-      'set-cookie'
-    ] as Array<string>
+    // previously
+
+    let headerSetCookies: Array<string> = []
+
+    try {
+      // returns headers.raw is not a function while running the sdk on a repl
+      headerSetCookies = (headers as any)?.raw()['set-cookie'] as Array<string>
+    } catch {
+      //  returns headers.get is not a function while running test cases
+      // according to the type definitions, .get happens to be the correct method to extract the cookie headers
+      headerSetCookies = headers
+        ?.get('set-cookie')
+        ?.split(';')
+        ?.map((headerValue) => String(headerValue).trim()) as Array<string>
```

Comment:
I find this type casting kind of weird. If it might be undefined, it seems dangerous to cast it to `Array<string>`, which makes the compiler unable to help us for type errors later on

Reply:
(by the way I know that a similar type cast was already happening for this variable which was not your fault!)